vault/resource_pki_secret_backend_root_cert.go

```diff
@@ -550,6 +540,7 @@ func pkiSecretBackendDeleteRootPath(backend string) string {
 	return strings.Trim(backend, "/") + "/root"
 }
 
+// TODO confirm if this can be removed since users should already be on 3.25
```

Comment:
Should these be removed? Since the upgrade guide instructs users to upgrade to 3.25 before the version bump, the users' TF states for this resource should already be at the latest schema version with the state upgraded. We can also keep this around for bookkeeping so that we don't forget the change that was made. Fine with whatever folks think is best

Reply:
Maybe we could mark it as deprecated to be removed in the next major version bump? This would be more of an "internal-only" deprecation and would be transparent to users so we wouldn't need to note it in any upgrade guide. Just some thoughts :)